internal/util/util.go

```diff
@@ -917,3 +918,17 @@ func FormatNumber(n int) string {
 		}
 	}
 }
+
+// YAMLMarshal with indent 2 (yaml.v3 indent default is 4)
+func MarshalYAML(v any) ([]byte, error) {
+	buf := bytes.Buffer{}
+	enc := yaml.NewEncoder(&buf)
+	defer enc.Close()
+
+	enc.SetIndent(2)
+	if err := enc.Encode(&v); err != nil {
+		return nil, err
+	}
+
+	return buf.Bytes(), nil
+}
```

Comment:
Was this the only breaking change from v3?

Also is unmarshaling behavior the same?

Maybe we should put this in its own package and have a marshal/unmarshal paired functions:

```
package serialization

func MarshalYaml(v any) ([]byte, error)

func UnmarshalYaml(in [][byte](https://pkg.go.dev/builtin#byte), out interface{}) (err [error](https://pkg.go.dev/builtin#error))
```

It may expand to other serialization techniques for other purposes.

Reply:
That's not all, there are some other changes. There's also the problem that [go-yaml/yaml](https://github.com/go-yaml/yaml) doesn't look like it's actively developed anymore, which means that the issues above probably can't be solved upstream:

* go-yaml/yaml#788
* go-yaml/yaml#776

So there are a couple of options right now

1. Revert to `yaml.v2`.
2. Use `yaml.v3` and follow the v3 format.
3. Switch to another yaml library (may have to follow their format too), such as:
  a. [goccy/go-yaml](https://github.com/goccy/go-yaml)
  b. [sigs.k8s.io/yaml](https://github.com/kubernetes-sigs/yaml)

I think in this PR we can select option 1 first, and then start a separate issue to handle this.

Reply:
see: Volte6/GoMud#280